app/javascript/components/catalog-form/catalog-form.schema.js

```diff
@@ -2,9 +2,9 @@ import { componentTypes } from '@data-driven-forms/react-form-renderer';
 import debouncePromise from '../../helpers/promise-debounce';
 import { API } from '../../http_api';
 
-export const asyncValidator = value => API.get(`/api/service_catalogs?expand=resources&filter[]=name=${value}`)
+export const asyncValidator = (value, id) => API.get(`/api/service_catalogs?expand=resources&filter[]=name=${value}`)
   .then((json) => {
-    if (json.resources.length > 0) {
+    if (json.resources.length > 0 && json.resources[0].id !== id) {
```

Comment:
What if the result is not first in the arrya? I think we should use `Array.prototype.find` for this.

Reply:
There should not be more than one result, even if a user injects `*` to the name of the catalog.

Reply:
are you 100% sure?

Reply:
With `find` that code is almost unbreakable with `[0]` a lot of things can go wrong in future.